cachix/src/Cachix/Client/Commands.hs

```diff
@@ -111,10 +113,10 @@ To instruct Nix to use the binary cache:
 
       |] :: Text)
 
-use :: ClientEnv -> Maybe Config -> Text -> Bool -> IO ()
-use env _ name shouldEchoNixOS = do
+use :: Env -> Text -> Bool -> IO ()
+use Env{..} name shouldEchoNixOS = do
```

Comment:
I see you're still doing this. The least you could do is use namedfieldpuns. It's still name shadowing, but at least we'll have a clear declaration site for the variable.

Reply:
Guilty as charged :) I wanted to refactor the code without too many changes - but somehow this backfired by disobeying the boy scout rule. I'll convert those to plain function calls.